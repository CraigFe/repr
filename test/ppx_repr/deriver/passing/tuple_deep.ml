(* Nested tuple type *)
type deep_tuple =
  (((int32 * int32) * int32 * int32) * int32 * int32) * int32 * int32
[@@deriving repr]

let (_ : deep_tuple Repr.Type.t) = deep_tuple_t
