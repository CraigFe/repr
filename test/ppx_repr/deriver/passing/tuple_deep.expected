type deep_tuple =
  ((((int32 * int32) * int32 * int32) * int32 * int32) * int32 * int32)
[@@deriving repr]
include
  struct
    let deep_tuple_t =
      Repr.Type.triple
        (Repr.Type.triple
           (Repr.Type.triple (Repr.Type.pair Repr.Type.int32 Repr.Type.int32)
              Repr.Type.int32 Repr.Type.int32) Repr.Type.int32
           Repr.Type.int32) Repr.Type.int32 Repr.Type.int32
  end[@@ocaml.doc "@inline"][@@merlin.hide ]
let (_ : deep_tuple Repr.Type.t) = deep_tuple_t
